vault/core.go

```diff
@@ -1516,3 +1516,27 @@ func (c *Core) BarrierKeyLength() (min, max int) {
 	max += shamir.ShareOverhead
 	return
 }
+
+func (c *Core) ValidateWrappingToken(token string) (bool, error) {
+	if token == "" {
+		return false, fmt.Errorf("token is empty")
+	}
+
+	te, err := c.tokenStore.Lookup(token)
+	if err != nil {
+		return false, err
+	}
+	if te == nil {
+		return false, nil
+	}
+
+	if te.Policies == nil || len(te.Policies) != 1 {
```

Comment:
Not unless you can guarantee it'll never, ever, under any circumstance ever, be `nil`, because an index on a `nil` slice is a panic and panics are always bugs :-)